### ISSUE TITLE
Collection should have generic type

### ISSUE THREAD

**Issue by jeff-tenhave** [Esri/jsapi-resources#78]
It would be very useful if Collection had a generic type variable. Example:
```
let collection = new Collection<string>(); 
collection.add(3);    // Fail at compile time
```
The current collection would be equivalent to Collection\<any\>.

Esri Dev Team: Would you be open to a PR with this change made to arcgis-js-api.d.ts?

**Comment by jkieboom** [comment]
The typings are automatically generated, so we will need to fix this on the generator or documentation side. It is not clear exactly when we will be able to do this.

**Comment by shenningsgard** [comment]
In the mean time, it *is* possible (although a bit fiddly) to fork and do this yourself. I made the same observation for 4.3 and made the changes to our own fork of the defs (along with some other fixes), and can confirm this makes it nicer to use Collection :)

**Comment by dasa** [comment]
Fixed in 4.5.
